File: frontend/src/types/piragi.ts
export interface DocumentSummary {
  category: string;
  file_count: number;
  path: string;
}

export interface ConnectDocumentsRequest {
  document_paths: string;
}

export interface ConnectDocumentsResponse {
  project_id: string;
  document_paths: string;
  connected: boolean;
}

export interface ChunkResult {
  chunk: string;
  source: string;
  relevance: number;
}

export interface PiragiQueryRequest {
  query: string;
  step_type: string;
}

export interface PiragiQueryResponse {
  query: string;
  results: ChunkResult[];
}

export interface PiragiDocumentsResponse {
  categories: DocumentSummary[];
}

export interface ConnectedDocuments {
  document_paths: string;
}

export interface UploadDocumentResponse {
  filename: string;
  category: string;
  path: string;
  size: number;
}

File: backend/app/schemas/piragi.py
from pydantic import BaseModel


class DocumentSummary(BaseModel):
    category: str
    file_count: int
    path: str


class ConnectDocumentsRequest(BaseModel):
    document_paths: str


class ConnectDocumentsResponse(BaseModel):
    project_id: str
    document_paths: str
    connected: bool


class ChunkResult(BaseModel):
    chunk: str
    source: str
    relevance: float


class PiragiQueryRequest(BaseModel):
    query: str
    step_type: str


class PiragiQueryResponse(BaseModel):
    query: str
    results: list[ChunkResult]


class PiragiDocumentsResponse(BaseModel):
    categories: list[DocumentSummary]


class UploadDocumentResponse(BaseModel):
    filename: str
    category: str
    path: str
    size: int
